src/User/Auth/authService/authService.js

```diff
@@ -0,0 +1,66 @@
+import AuthUser from "../AuthUser.js"
+import bcrypt from "bcrypt";
+import { v4 as uuidv4 } from 'uuid';
+import mailService from "./mailService.js";
+import Post from "../../../Post/Post.js";
+
+class AuthService {
+    async registration(username, password, email) {
+        const newUser = await AuthUser.findOne({ username })
+        const newEmail = await AuthUser.findOne({ email })
+        if (newUser || newEmail) {
+            throw new Error('such user exists')
+        }
+        const hashPassword = await bcrypt.hash(password, 5)
```

Comment:
The next step will be adding salt and usage sha256 algorithm. For now leave it as it